### PR TITLE
Add exclusive control to `File.contents` and `File.lines`

### PR DESCRIPTION
**Pull request by norio-nomura**
- Fix https://github.com/realm/SwiftLint/issues/2089
- Remove `File.lines:setter`

### CODE REVIEW

**Comment by jpsim** [comment]
👍 with a changelog entry

File: Source/SourceKittenFramework/File.swift [hunk 1]
```diff
@@ -16,36 +17,46 @@ import SWXMLHash
...
+                if _contents == nil {
+                    _contents = try! String(contentsOfFile: path!, encoding: .utf8)
+                }
             }
             return _contents!
         }
         set {
-            _contents = newValue
+            _contentsQueue.sync {
+                _contents = newValue
```
Comment (segiddins) [inline Source/SourceKittenFramework/File.swift]: should this also invalidate `_lines`?

Comment (norio-nomura) [reply Source/SourceKittenFramework/File.swift]: You are right.
I'll make additional changes.

Comment (jpsim) [reply Source/SourceKittenFramework/File.swift]: Nice catch Sam 👍

**Comment by norio-nomura** [comment]
Could you please review again?

**Pull request merged**